### DISCUSSION TITLE
Accessibility problem for blind musician on MAC version

### DISCUSSION THREAD

**Discussion by Fandango64** [jamulussoftware/jamulus#2763] [Technical Support & Help]
Hello. I am a MAC blind Jamulus user, I am writing to report that we blind users on MAC Os. we have great difficulties in using the software to the lack of accessibility with the Apple Voice Over screen reader. There are many problems with buttons and combo box, in addiction the server list is almost completely inaccessible and it is very difficult to select one. I have been able to see the functioning of the Windows version which is much more accessible with the screen reader.
I ask if it was possible for the next releases to implement accessibility for blind musicians in the MAC version as well.
Finally I point out that, I had a notification of a new update but for accessibility problems I did not know if the version of is updated or not.

Many thanks for what you can do, I am available for any tests.

Many thanks

**Comment by pljones** [comment]
Hi,

We use the Qt standard controls for all platforms with nothing special done to enable accessibility features on Windows to the best of my knowledge.

So I'm not sure what would need changing on MacOS.  My understanding is that MacOS support may generally be improved by the move to a newer version of Qt but that will take some time.

If anyone can offer assistance here it would be appreciated.

Thanks,

-- Peter

**Reply by ann0see** [reply]
@chigkim might be able help here.

**Reply by Fandango64** [reply]
Hello

thanks everyone for the answers and thanks Chigkim

  for your work on Jamulus accessibility, I downloaded the file but unfortunately my MAC updates only arrive at High Sierra 10.13.6, is there a version running also on High sierra?

**Reply by pljones** [reply]
The other builds are on @chigkim's repository (which needs updating, as it says it's 3.9.0, but it isn't).
https://github.com/chigkim/jamulus/actions/runs/2768253539
Pick the macos legacy build and unzip the file to get the unsigned dmg.

**Reply by Fandango64** [reply]
Hello 
many thanks i downloaded the legacy version which worked perfectly on my old mac book pro from 2010! :-)

Good music.

**Comment by chigkim** [comment]
Most of my accessibility fixes are incorporated into r3_9_0, but the server list is only partially accessible on Mac due to a QT bug.
Check out my accessible build. https://chigkim.github.io/jamulus/
There are many VoiceOver users including myself who use Jamulus successfully on Mac.

**Reply by pljones** [reply]
Is [this change](https://github.com/jamulussoftware/jamulus/compare/master...chigkim:jamulus:master) ready to contribute, do you reckon?

**Comment by chigkim** [comment]
I think it can't be merged into the main Jamulus because I believe it'll 
change the GUI for sighted users.

As a hackaround, I had to make each ping time, server name, as separate 
buttons on server list for VoiceOver to be able to navigate.

QT has to fix the treview on Mac for VoiceOver.

**Comment by chigkim** [comment]
It should be r3_9_0 on https://chigkim.github.io/jamulus/.

**Reply by pljones** [reply]
You should avoid releasing with the official release number.  It will confuse the upgrades.  Always wait until the version has been rolled forwards to the next dev release before updating.  And try to avoid working on master even in your own repo - it's a bad habit.

**Comment by chigkim** [comment]
As soon as official release comes out, people bug me to update, so I can't wait.
Also I name it something like Jamulus_Accessible_Mac_3_9_0 to differentiate from the main depo releases and I tag it with alpha.1,2,3...

**Reply by pljones** [reply]
OK, so long as the `.dmg` name is different I guess it's fine.
